app/src/main/java/org/mozilla/focus/Components.kt

```diff
@@ -155,6 +157,7 @@ class Components(
 
         val exceptionsMigrator = EngineProvider.provideTrackingProtectionMigrator(context)
         exceptionsMigrator.start(context)
+        StrictMode.setThreadPolicy(threadPolicy)
```

Comment:
We could also take this off the main thread, I guess? Reading the shared preference is thread-safe and `setPermission` has an `@AnyThread` annotation. Only the engine needs to be created on the main-thread, but since we are calling `components.engine.warmUp()` in `FocusApplication` before running this migration, that has already happened.

Reply:
I'm getting exception from getStorageController() when running out of Main :( 

```
    java.lang.IllegalThreadStateException: Expected thread 2 ("main"), but running on thread 15965 ("DefaultDispatcher-worker-3")
        at org.mozilla.gecko.util.ThreadUtils.assertOnThreadComparison(ThreadUtils.java:110)
        at org.mozilla.gecko.util.ThreadUtils.assertOnThread(ThreadUtils.java:86)
        at org.mozilla.gecko.util.ThreadUtils.assertOnUiThread(ThreadUtils.java:77)
        at org.mozilla.geckoview.GeckoRuntime.getStorageController(GeckoRuntime.java:855)
        at org.mozilla.focus.exceptions.GeckoExceptionsMigrator.migrate(GeckoExceptionsMigrator.kt:56)
        at org.mozilla.focus.exceptions.GeckoExceptionsMigrator.start(GeckoExceptionsMigrator.kt:27)
        at org.mozilla.focus.Components.migrateTrackingProtectionExceptions(Components.kt:158)
        at org.mozilla.focus.FocusApplication$onCreate$1.invokeSuspend(FocusApplication.kt:62)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

Reply:
I changed a bit the order in which we are getting the `storageController` and just take `start` of the main thread. 

https://github.com/mozilla-mobile/focus-android/pull/4977/files#diff-af021628211199f43ada47c8919e6cb79a1205a19ed9abbe349a2d01108cfd44R161